Sources/XiEditor/MarkerBar.swift

```diff
@@ -69,18 +69,21 @@ class MarkerBar: NSScroller {
 
     func drawMarkers() {
         let path = CGMutablePath()
-        let totalLines = CGFloat(parent.lines.height)
-        let markerBarHeight = self.layer?.bounds.height
-        let maxScrollerWidth = self.layer?.bounds.width
+        let totalLines = CGFloat(markerDelegate!.lines.height)
+
+        print(totalLines)
+
+        guard let markerBarHeight = layer?.bounds.height,
```

Comment:
It's idiomatic to put the guard statement at the very top of the function; that way when reading the function the first thing you do is scan the `guard`, which tells you about the preconditions the function expects to hold.